### DISCUSSION TITLE
Previously created local HTML reports not working after 0.9.0 release

### DISCUSSION THREAD

**Discussion by skeletozaure** [datapane/datapane#56] [General]
Hello,

I'm using Datapane to generate a report, then i send it by email to some users. It was working  fine, till today when a first user told me that the report was displayed as blank in its browser (Chrome). And after checking it's the same for me and all other users. I've tried Chrome, Edge, even Internet Explorer, and the report is no more displayed, even if source code of the page is OK. 

I have joined the file as a zip file below :

[2020-12-30-SSISCheck.zip](https://github.com/datapane/datapane/files/5763589/2020-12-30-SSISCheck.zip)

Opening debugger in Chrome, here are the errors i'm seeing : 

report-base.js:1 Error: Can't deserialize Blocks at page level: must be Group or Select
    at report-base.js:1
    at Array.forEach (<anonymous>)
    at f.deserializePage (report-base.js:1)
    at report-base.js:1
    at Array.forEach (<anonymous>)
    at f.deserialize (report-base.js:1)
    at u.xml2json (report-base.js:1)
    at u.load (report-base.js:1)
    at report-base.js:1
    at as (report-base.js:1)

It seems that there is something weird in report-base.js. did you made some updates to the back end ?

Thanks for your help,

Cédric

**Comment by lanthias** [comment]
Hi Cedric, thank a lot for your note.

We updated datapane, and as such older reports which were not generated in standalone mode will need to be regenerated after updating the library (`pip3 install -U datapane`)

As part of this update, we have added a feature which will mean that you don't need to do this in the future -- i.e. old generated reports which pull down javascript from our CDN will pull down the appropriate version, and not the latest version (which may be incompatible).

Let me know if this sorts it for you!

Best
Leo

**Comment by skeletozaure** [comment]
Hello Leo,

Fantastic, it worked for me, users are happy again :) Let me congrat you
for Datapane which is absolutely unique and so easy to use. I'm a great
fan, and really want to make it used in our plant more widely that just my
experiment.

have a nice day,
Cédric

**Comment by mands** [comment]
Thanks for the kind words Cédric!

Just to add to what @lanthias wrote, when saving your reports, you can use the `standalone` argument, e.g. `report.save("report.html", standalone=True)` - this will embed all the code required to run the report in the HTML file, meaning that those reports will always keep working (however each report will be a _lot_ larger so it depends on your use-case). However the changes we've just made also mean that your reports won't break going forwards either way.

Cheers!
Mandeep

**Comment by mands** [comment]
Also referenced in #62

**Comment by skeletozaure** [comment]
Thanks Mandeep for this nice precision.

Cédric
